ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/explorer/project/ProjectExplorerPresenter.java

```diff
@@ -438,6 +443,15 @@ public boolean isShowHiddenFiles() {
         return hiddenFilesAreShown;
     }
 
+    @Override
+    public void onVcsBranchClicked(Project project) {
+        showVcsBranchActionProviders.get()
+                                    .stream()
+                                    .filter(provider -> provider.getVcsName().equals(project.getAttribute("vcs.provider.name")))
```

Comment:
I would recommend to extract `project.getAttribute("vcs.provider.name")` to calculate it only once instead of for each action provider in a stream, what do you think?